explorer/mquery.go

```diff
@@ -324,33 +324,53 @@ func (v *Impact) Merge(base *Impact) {
 	if base == nil {
 		return
 	}
+
 	if v.Scoring == Impact_SCORING_UNSPECIFIED {
 		v.Scoring = base.Scoring
 	}
+	if v.Value == nil {
+		v.Value = base.Value
+	}
+	if v.Weight < 1 {
+		v.Weight = base.Weight
+	}
 }
 
 func (v *Impact) UnmarshalJSON(data []byte) error {
 	var res int32
-	v.Value = -1
-	v.Weight = -1
+	if err := json.Unmarshal(data, &res); err == nil {
+		v.Value = &ImpactValue{Value: res}
+		return nil
+	}
 
+	type tmp Impact
+	return json.Unmarshal(data, (*tmp)(v))
```

Comment:
@jaym this part is still necessary for straightforward unmarshaling e.g. `impact: 70` in yaml. The complex struct (impact value + weight + scoring) is an advanced variant with more fine-grained control.